api/peer/transport.go

```diff
@@ -25,7 +25,7 @@ package peer
 // Subscriber listens to changes of a Peer over time.
 type Subscriber interface {
 	// The Peer Notifies the Subscriber when its status changes (e.g. connections status, pending requests)
-	NotifyStatusChanged(Identifier)
+	NotifyStatusChanged(Identifier) error
```

Comment:
It looks like this change is bubbling up the `error` return all the way up to
Peer, Chooser, etc.

The way I understand this API is, it's an asynchronous out-of-band
notification to the subscriber that something happened. In fact, I'm fairly
sure this will be run inside or fork a goroutine in certain scenarios.
Failures in this are not really the caller's problem. The implementation can
log these failures if necessary but these should not cause the notifier to
crash.